GCRCatalogs/dc2_object.py

```diff
@@ -185,11 +188,18 @@ def _subclass_init(self, **kwargs):
             err_msg = 'No catalogs were found in `base_dir` {}'
             raise RuntimeError(err_msg.format(self.base_dir))
 
-        self._columns = self._generate_columns(self._datasets)
+        if self._schema_path is None or not os.path.exists(self._schema_path):
```

Comment:
See my comment below on Line 400, and consider changing the logic here to something like:
```python
        self._columns = None
        if self._schema_path and os.path.exists(self._schema_path):
            self._columns = self._generate_columns_from_yaml(self._schema_path)
        if not self._columns:
            warn_msg = 'No schema file found: "{}".\nFalling back to reading all datafiles for column names'
            warnings.warn(warn_msg.format(self._schema_path))
            self._columns = self._generate_columns(self._datasets)
```